Document type inference issues with dynamic params in SQL (#10801)
apache/druid: docs/querying/sql.md

corresponding parameter value when you execute the query. Parameters are bound to the placeholders in the order in
which they are passed. Parameters are supported in both the [HTTP POST](#http-post) and [JDBC](#jdbc) APIs.

> Please note that in certain cases, dynamic parameters used in expressions can cause type inference issues which can cause the query to fail. The following query is one such case:

Reviewer: ```suggestion
In certain cases, using dynamic parameters in expressions can cause type inference issues which cause your query to fail, for example:
```
Avoid "please note". Also does the type interface issue result in a specific error? If so might be nice to include it.

```sql
SELECT * FROM druid.foo WHERE dim1 like CONCAT('%', ?, '%')
```

> In these cases, the solution would be to explicitly provide the type of the dynamic parameter using the `CAST` keyword. The above example can be fixed as:

Reviewer: ```suggestion
To solve this issue, explicitly provide the type of the dynamic parameter using the `CAST` keyword. Consider the fix for the preceding example:
```
